### PR TITLE
Assignment-4 | Material-UI |Form Validation |Controlled Components |used YUP and FORMIK for form validation

### PR DESCRIPTION
**Pull request by kaushalyad321**

### CODE REVIEW

File: Assignment-4/app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import DropDownMenu from "./Components/DropDownMenu";
+import ToDoList from "./Components/ToDoList";
+import TempConverter from "./Components/TempConverter";
+import FormValidation from "./Components/FormValidation";
+import FormSubmitOrNot from "./Components/FormSubmitOrNot";
+import FormValidationRule from "./Components/FormValidationRule";
+import DataTable from './Components/DataTable'
+import SidebarNavigation from "./Components/SidebarNavigation";
+import FilterComponent from "./Components/FilterComponent";
+import ShowModel from './Components/ShowModel'
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/App.js]: Create a index file and import all the components from there

File: Assignment-4/app/src/Components/SpecificInputValue.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const [massage, setMassage] = useState("");
+  const handleClick = () => {
+    if (inputValue === "show") {
+      setMassage("well done friend your input value is correct ");
+    }
+  };
+
+  let result = "";
+  if (massage.length > 0) {
+    result = massage;
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/SpecificInputValue.js]: Task 3 not working

File: Assignment-4/app/src/Components/CheckPassword.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+
+const CheckPassword = () => {
+  const [password, setPassword] = useState("");
+  const [rePassword, setRePassword] = useState("");
+  const [result, setResult] = useState("");
+
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/CheckPassword.js:7]: Not working as expected

File: Assignment-4/app/src/App.js (hunk 1 above)
Comment (kaushalyad321) [reply Assignment-4/app/src/App.js]: ok

File: Assignment-4/app/src/Components/SpecificInputValue.js (hunk 2 above)
Comment (kaushalyad321) [reply Assignment-4/app/src/Components/SpecificInputValue.js]: if we put show input then only massage will display

File: Assignment-4/app/src/Components/TempConverter.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React, { useState } from "react";
+const TempConverter = () => {
+  const [tempCel, setTempCel] = useState(0);
+  const [tempFah, setTempFah] = useState(32);
+  const fohrenheitController = (e) => {
+    setTempFah(e.target.value);
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/TempConverter.js:7]: image
Giving incorrect result

File: Assignment-4/app/src/Components/FilterComponent.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    "drt",
+    "uytr",
+    "wqer",
+    "rewt",
+    "hjsj",
+    "kiu",
+    "qwert",
+    "name",
+    "pappu",
+    "kaushal",
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/FilterComponent.js:21]: put some meaningful name in list

Comment (kaushalyad321) [reply Assignment-4/app/src/Components/FilterComponent.js:21]: ok

File: Assignment-4/app/src/Components/SidebarNavigation.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  const toggleSidebar = () => {
+    setSidebarOpen(!isSidebarOpen);
+  };
+
+  return (
+    <div>
+      <header>
+        <button onClick={toggleSidebar}>
+          <img
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/SidebarNavigation.js]: image
All these links navigating to home page only

File: Assignment-4/app/src/Components/FormValidation.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  const [formData, setFormData] = useState({
+    name: "",
+    email: "",
+  });
+
+  const [errors, setErrors] = useState({
+    name: "",
+    email: "",
+  });
+
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/FormValidation.js:23]: task 10 not working as expected

File: Assignment-4/app/src/Components/ShowModel.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+  const [isDialogOpen, setDialogOpen] = useState(false);
+
+  const handleOpenDialog = () => {
+    setDialogOpen(true);
+  };
+
+  const handleCloseDialog = () => {
+    setDialogOpen(false);
+  };
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/ShowModel.js:15]: We need to discuss this

File: Assignment-4/app/src/Components/DataTable.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  { id: 7, name: "Golu", age: 33, city: "Noida" },
+  { id: 8, name: "Chhero", age: 11, city: "Noida" },
+  { id: 9, name: "Tuna", age: 27, city: "Noida" },
+  { id: 10, name: "Keko", age: 29, city: "Noida" },
+  { id: 11, name: "Nanu", age: 20, city: "Noida" },
+  { id: 12, name: "Thelo", age: 12, city: "Noida" },
+];
+
+const columns = [
+  { id: "name", label: "Name" },
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/DataTable.js]: Not working correctly

File: Assignment-4/app/src/Components/DataTable.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+  { id: "city", label: "City" },
+];
+
+const DataTable = () => {
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(5);
+  const [orderBy, setOrderBy] = useState(columns[0].id);
+  const [order, setOrder] = useState("asc");
+
+  const handleChangePage = (event, newPage) => {
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/DataTable.js]: sorting on name column not working

File: Assignment-4/app/src/Components/FormValidationRule.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+      .min(10, "mobile no is not valid")
+      .max(10, "mobile no is not valid!")
+      .required("mobile no is Required"),
+    email: yup.string().email("Invalid email").required("Email is Required"),
+  });
+
+  const handleSubmit = () => {
+    console.log(formFields);
+  };
+
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/FormValidationRule.js:31]: image
email validation is not correct

Comment (kaushalyad321) [reply Assignment-4/app/src/Components/FormValidationRule.js:31]: ok fixing

File: Assignment-4/app/src/Components/ShowModel.js (hunk 8 above)
Comment (kaushalyad321) [reply Assignment-4/app/src/Components/ShowModel.js:15]: ok

File: Assignment-4/app/src/Components/SidebarNavigation.js (hunk 6 above)
Comment (kaushalyad321) [reply Assignment-4/app/src/Components/SidebarNavigation.js]: please check i fixed the issue

File: Assignment-4/app/src/Components/FilterComponent.js (hunk 5 above)
Comment (kaushalyad321) [reply Assignment-4/app/src/Components/FilterComponent.js:21]: updated meaningful name

File: Assignment-4/app/src/Components/TempConverter.js (hunk 4 above)
Comment (kaushalyad321) [reply Assignment-4/app/src/Components/TempConverter.js:7]: please check fixed the issues

File: Assignment-4/app/src/Components/DropDownMenu.js [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+    return(
+    <div>
+      <form>
+        <select onClick={(e)=>{setOption(e.target.value)}}>
+          <option value="" selected disabled>
+            Select a Color
+          </option>
+          <option value="blue">Blue</option>
+          <option value="green">Green</option>
+          <option value="red" style={{}}>
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/DropDownMenu.js]: Remove this unused style

Comment (kaushalyad321) [reply Assignment-4/app/src/Components/DropDownMenu.js]: ok

File: Assignment-4/app/src/Components/FilterComponent.js [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+
+  return (
+    <>
+      <div>
+        <input
+          onChange={(e) => {
+            setInputValue(e.target.value);
+          }}
+        ></input>
+        <button>search</button>
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/FilterComponent.js:65]: there's no use of this search button here

Comment (kaushalyad321) [reply Assignment-4/app/src/Components/FilterComponent.js:65]: just made for looking good

File: Assignment-4/app/src/Components/ModalDialog.js [hunk 14]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+
+const ModalDialog = ({ open, onClose, title, content, onSubmit }) => {
```
Comment (sumitbaudh22) [inline Assignment-4/app/src/Components/ModalDialog.js:8]: What is the use of this component if not any use then remove this file

**Pull request merged**